File: problemas/1100/1134.dart
import 'dart:io';

void main() {
  final combustiveis = <int, String> {
    1: 'Alcool',
    2: 'Gasolina',
    3: 'Diesel',
    4: 'Fim'
  };

  int endKey = combustiveis.keys.firstWhere((x) => combustiveis[x] == 'Fim');
  var abastecidos = new Map<int, int>();
  int tipo;

  combustiveis.forEach((key, value) {
    abastecidos.addAll({key : 0});
  });

  while ((tipo = int.parse(stdin.readLineSync())) != endKey) {
    if (abastecidos.containsKey(tipo)) {
      abastecidos[tipo]++;
    }
  }

  abastecidos.remove(endKey);
  stdout.writeln('MUITO OBRIGADO');
  abastecidos.forEach((key, value) {
    stdout.writeln('${combustiveis[key]}: ${value}');
  });
}

File: problemas/1100/1134.php
<?php

$combustiveis = [
    1 => 'Alcool',
    2 => 'Gasolina',
    3 => 'Diesel',
    4 => 'Fim'
];

$endKey = array_search('Fim', $combustiveis);
$abastecidos = array();

foreach ($combustiveis as $key => $value) {
    $abastecidos += array($key => 0);
}

while (($tipo = readline()) != $endKey) {
    if (array_key_exists($tipo, $abastecidos)) {
        $abastecidos[$tipo]++;
    }
}

unset($abastecidos[$endKey]);
echo "MUITO OBRIGADO\n";
foreach ($abastecidos as $key => $value) {
    echo $combustiveis[$key] . ': ' . $value . "\n";
}